### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by Cyber1000**
Information about this PR:
- there are only small changes in old code (some return values, to make it better testable)
- this adds some classes for pytest
   - http-client/cli is tested with a small flask (http) server, simulating the device
   - both coap-clients/cli are tested with a coap server created by CoAPthon3 pip
      - I couldn't test the problem with very long answers (https://github.com/rgerganov/py-air-control/issues/35#issuecomment-631899965) cause between a CoAPthon3-client and a CoAPthon3-server it seems to work, maybe the problem is a bad coap-implementation of the device itself.
   - I'm testing client and cli were appropriate
- **I've removed python 3.4 support**, cause in coap_client.py I'm using <bytestring>.hex() (already merged this way in your master) and hex was added in python 3.5 . I found this problem while testing (help would work, but get and set wouldn't)
   - If you still want 3.4, I'll try to find a workaround
   - If not, I should adjust README, before merging
- I've added 3 TODOs (search "# TODO"), I don't know really if I should let them in, TODOs in Code tend to be forgotten, but I hope that PRs in the near future make them unnecessary
- plain_coap still uses test data from a coap device (I think they should be quite similar from the kind of output, but I don't know exactly), we may wait for #43 before merging

### CODE REVIEW

**Comment by rgerganov** [comment]
Thank you for your work. I will go through this next week.

File: pyairctrl/airctrl.py [hunk 1]
```diff
@@ -422,7 +422,6 @@ def main():
     for device in devices:
         if args.protocol == 'http':
             c = HTTPAirCli(device['ip'])
-            c.load_key()
```
[the remark this answers is not in the document]
Comment (Cyber1000) [reply pyairctrl/airctrl.py:425]: good catch, missed that ...

File: .travis.yml [hunk 2]
```diff
@@ -1,11 +1,16 @@
 language: python
 python:
-    - "3.4"
```
[the remark this answers is not in the document]
Comment (Cyber1000) [reply .travis.yml]: thanks for the information, works again with python 3.4!

File: .travis.yml [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
     - "3.5"
     - "3.6"
     - "3.7"
     - "3.8"
 install:
     - pip install -r requirements.txt
     - pip install .
-script: airctrl -h && cloudctrl -h
+    - pip install coveralls flask pytest
+    - sudo iptables -t nat -I OUTPUT -p tcp -o lo --dport 80 -j REDIRECT --to-ports 5000
```
[the remark this answers is not in the document]
Comment (Cyber1000) [reply .travis.yml:12]: plain_coap and coap take port as init-argument, you are right, we should implement this too with http, but I didn't want to change too much for now.

I think we should do this in a separate PR.

**Comment by Cyber1000** [comment]
I've adjusted everything, except the iptables-hack (see the comment)
Thanks for reviewing!

**Pull request merged**